### ISSUE TITLE
Microsoft defender false positives of Go binaries on windows

### ISSUE THREAD

**Issue by 42wim** [microsoft/go#1255]
Go binaries are more and more detected as false positive trojans by the (machine learning AI) of Microsoft Defender.

This makes it very difficult for developers to support and release binaries for windows.
(Compiling the same code as a linux binary btw doesn't get detected by Defender)

See also 
- https://go.dev/doc/faq#virus
- https://www.reddit.com/r/golang/comments/s1bh01/goexecutables_and_windows_defender/
- https://github.com/kachick/dotfiles/issues/442
- multiple other sources on github/internet

This isn't probably that can be solved here, but I didn't really find a better way to contact anyone about this.

**Comment by dagood** [comment]
Yeah, on the Microsoft Go team we're aware of Windows Security/Defender issues with Go apps. You're right that we can't exactly go fix something and solve all Go false positives. 😄 I do have a few links about this to share, though:

There is some information here about some steps you can take as a developer: [Partnering with the industry to minimize false positives](https://www.microsoft.com/en-us/security/blog/2018/08/16/partnering-with-the-industry-to-minimize-false-positives/). The headliner is signing. Signing is already mentioned in the Reddit post you linked, but this article explains a bit more about why that helps. The other tips are also interesting, but whether or not they're helpful probably depends on the app.

Ultimately, you can also report individual false positives to https://www.microsoft.com/en-us/wdsi/filesubmission.

**Comment by 42wim** [comment]
Thanks for the feedback, I'm already signing the executables with sectigo and unfortunately my experience with signing is that it doesn't really make much of a difference. It seems random.

**Comment by ksylvan** [comment]
Yes. I have this problem with almost ALL the package patches to winget-pkgs, like this one:

https://github.com/microsoft/winget-pkgs/pull/292992

**Comment by dagood** [comment]
I see from that thread that you submitted https://www.microsoft.com/en-us/wdsi/submission/b662939b-1edc-46cd-ab37-40d6125380cf, have you made any more submissions? I don't know if there's much we can do to follow up on this, but concrete examples like that are the best we can hope for.

**Comment by ksylvan** [comment]
Thanks for the response. I have many links in all my merged/closed PRs here @dagood 

https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+is%3Aclosed+author%3Aksylvan

For example, in https://github.com/microsoft/winget-pkgs/pull/289181 we have the following:

https://www.microsoft.com/en-us/wdsi/submission/5d0490ad-2352-46c3-a851-4102458240db

In https://github.com/microsoft/winget-pkgs/pull/291424 we have: https://www.microsoft.com/en-us/wdsi/submission/3a706dfd-a3e7-40fe-b613-ab6e0fc3f10e

etc. There's at least one submission for each new version.

**Comment by ksylvan** [comment]
And when I tried adding `garble` to the mix, it then got flagged like this:

"a variant of WinGo/Packed.Obfuscated.D suspicious application"

https://github.com/microsoft/winget-pkgs/pull/294021

**Comment by dagood** [comment]
For what it's worth, I would expect garble to increase false positives. The scanner's trying to find certain patterns, so obfuscation with garble increases scrutiny rather than avoiding it. I tried garble a bit ago, had similar results internally, and ended up not using it for now.
